### DISCUSSION TITLE
Does goyave has swagger or any kinda api documentation feature in it?

### DISCUSSION THREAD

**Discussion by Bishwas-py** [go-goyave/goyave#232] [Q&A] [answered]
I am curious if I can have any API documentation in goyave.

**Comment by System-Glitch** [comment, marked as answer]
It used to. v4 had an unfinished [open API generator](https://github.com/go-goyave/openapi3) that was discontinued with v5. 

In the [v5 release notes](https://github.com/go-goyave/goyave/releases/tag/v5.0.0), it is said:
> The development of the framework will now consider real-world use-cases and scenarios with more importance. No "quick and easy" solutions, tools or fancy gadgets will be developed from now on.

The reason it has been dropped is because in a real-world medium-sized project, it is often preferred to maintain such documentation manually. A generator can only get you the basic route information (and at most the request validation) with so many missing pieces that you would end up writing more than half of it in annotations. The system gathering information from code comments was broken with go 1.18 (https://github.com/go-goyave/openapi3/commit/7881a09c1ed0cc682b10290d901e3df53986a651).

In that case, it is preferable to directly write this documentation in the OpenAPI yml spec files. This way you will get all the flexibility that OpenAPI can provide without the inconveniences brought by the generator.
